// @prisma/client/index-browser.js

/* !!! This is code generated by Prisma. Do not edit directly. !!!
/* eslint-disable */
// biome-ignore-all lint: generated file

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal,
  DbNull,
  JsonNull,
  AnyNull,
  NullTypes,
  makeStrictEnum,
  Public,
  getRuntime,
  skip
} = require('./runtime/index-browser.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 7.8.0
 * Query Engine version: 3c6e192761c0362d496ed980de936e2f3cebcd3a
 */
Prisma.prismaVersion = {
  client: "7.8.0",
  engine: "3c6e192761c0362d496ed980de936e2f3cebcd3a"
}

Prisma.PrismaClientKnownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientKnownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientUnknownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientRustPanicError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientRustPanicError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientInitializationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientInitializationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientValidationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientValidationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`sqltag is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.empty = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`empty is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.join = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`join is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.raw = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`raw is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.getExtensionContext is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.defineExtension = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.defineExtension is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = DbNull
Prisma.JsonNull = JsonNull
Prisma.AnyNull = AnyNull

Prisma.NullTypes = NullTypes



/**
 * Enums
 */

exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.OrganizationScalarFieldEnum = {
  id: 'id',
  name: 'name',
  slug: 'slug',
  isActive: 'isActive',
  enabledModules: 'enabledModules',
  subscriptionPlan: 'subscriptionPlan',
  country: 'country',
  currency: 'currency',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.BrandScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  name: 'name',
  slug: 'slug',
  domain: 'domain',
  themeColors: 'themeColors',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.UserScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  email: 'email',
  passwordHash: 'passwordHash',
  name: 'name',
  role: 'role',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.HotelScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  brandId: 'brandId',
  name: 'name',
  address: 'address',
  isActive: 'isActive',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.RoomScalarFieldEnum = {
  id: 'id',
  hotelId: 'hotelId',
  roomNumber: 'roomNumber',
  type: 'type',
  dailyRate: 'dailyRate',
  capacity: 'capacity',
  status: 'status',
  isHourlyAvailable: 'isHourlyAvailable',
  rate3h: 'rate3h',
  rate6h: 'rate6h',
  rate9h: 'rate9h',
  rate12h: 'rate12h',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.BookingScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  roomId: 'roomId',
  guestId: 'guestId',
  guestName: 'guestName',
  guestEmail: 'guestEmail',
  guestPhone: 'guestPhone',
  startTime: 'startTime',
  endTime: 'endTime',
  totalPrice: 'totalPrice',
  status: 'status',
  notes: 'notes',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.WebsiteScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  brandId: 'brandId',
  title: 'title',
  description: 'description',
  subdomain: 'subdomain',
  customDomain: 'customDomain',
  status: 'status',
  config: 'config',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.CMSPageScalarFieldEnum = {
  id: 'id',
  websiteId: 'websiteId',
  title: 'title',
  slug: 'slug',
  isHome: 'isHome',
  seoTitle: 'seoTitle',
  seoDescription: 'seoDescription',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.CMSSectionScalarFieldEnum = {
  id: 'id',
  pageId: 'pageId',
  type: 'type',
  content: 'content',
  order: 'order',
  isActive: 'isActive'
};

exports.Prisma.RestaurantScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  hotelId: 'hotelId',
  name: 'name',
  description: 'description',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deletedAt: 'deletedAt'
};

exports.Prisma.TableScalarFieldEnum = {
  id: 'id',
  restaurantId: 'restaurantId',
  number: 'number',
  capacity: 'capacity',
  status: 'status'
};

exports.Prisma.MenuScalarFieldEnum = {
  id: 'id',
  restaurantId: 'restaurantId',
  name: 'name',
  isActive: 'isActive'
};

exports.Prisma.MenuCategoryScalarFieldEnum = {
  id: 'id',
  menuId: 'menuId',
  name: 'name',
  order: 'order'
};

exports.Prisma.MenuItemScalarFieldEnum = {
  id: 'id',
  categoryId: 'categoryId',
  name: 'name',
  description: 'description',
  price: 'price',
  image: 'image',
  isAvailable: 'isAvailable'
};

exports.Prisma.OrderScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  restaurantId: 'restaurantId',
  tableId: 'tableId',
  bookingId: 'bookingId',
  type: 'type',
  status: 'status',
  totalPrice: 'totalPrice',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.OrderItemScalarFieldEnum = {
  id: 'id',
  orderId: 'orderId',
  menuItemId: 'menuItemId',
  quantity: 'quantity',
  price: 'price',
  notes: 'notes'
};

exports.Prisma.InventoryItemScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  name: 'name',
  sku: 'sku',
  unit: 'unit',
  quantity: 'quantity',
  minQuantity: 'minQuantity',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.InventoryTransactionScalarFieldEnum = {
  id: 'id',
  itemId: 'itemId',
  type: 'type',
  quantity: 'quantity',
  reference: 'reference',
  createdAt: 'createdAt'
};

exports.Prisma.StaffScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  userId: 'userId',
  designation: 'designation',
  salary: 'salary',
  joiningDate: 'joiningDate',
  status: 'status'
};

exports.Prisma.AttendanceScalarFieldEnum = {
  id: 'id',
  staffId: 'staffId',
  date: 'date',
  clockIn: 'clockIn',
  clockOut: 'clockOut',
  status: 'status'
};

exports.Prisma.GuestProfileScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  name: 'name',
  email: 'email',
  phone: 'phone',
  preferences: 'preferences',
  loyaltyPoints: 'loyaltyPoints',
  isKswProfile: 'isKswProfile',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.InvoiceScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  bookingId: 'bookingId',
  orderId: 'orderId',
  parlorBookingId: 'parlorBookingId',
  invoiceNumber: 'invoiceNumber',
  totalAmount: 'totalAmount',
  tax: 'tax',
  status: 'status',
  billingSnapshot: 'billingSnapshot',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ExpenseScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  category: 'category',
  amount: 'amount',
  description: 'description',
  date: 'date'
};

exports.Prisma.ActivityLogScalarFieldEnum = {
  id: 'id',
  adminId: 'adminId',
  adminName: 'adminName',
  action: 'action',
  targetType: 'targetType',
  targetId: 'targetId',
  details: 'details',
  createdAt: 'createdAt'
};

exports.Prisma.ParlorCategoryScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  name: 'name',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ParlorServiceScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  categoryId: 'categoryId',
  name: 'name',
  description: 'description',
  price: 'price',
  duration: 'duration',
  image: 'image',
  isActive: 'isActive',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ServiceBookingScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  bookingId: 'bookingId',
  guestName: 'guestName',
  guestEmail: 'guestEmail',
  guestPhone: 'guestPhone',
  bookingTime: 'bookingTime',
  status: 'status',
  paymentMethod: 'paymentMethod',
  paymentStatus: 'paymentStatus',
  totalPrice: 'totalPrice',
  notes: 'notes',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.BookingServiceItemScalarFieldEnum = {
  id: 'id',
  bookingId: 'bookingId',
  serviceId: 'serviceId',
  priceAtBooking: 'priceAtBooking'
};

exports.Prisma.PlatformUserScalarFieldEnum = {
  id: 'id',
  username: 'username',
  email: 'email',
  phone: 'phone',
  passwordHash: 'passwordHash',
  isVerified: 'isVerified',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.OrganizationMembershipScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  platformUserId: 'platformUserId',
  role: 'role',
  joinedAt: 'joinedAt',
  status: 'status'
};

exports.Prisma.CustomerScalarFieldEnum = {
  id: 'id',
  phone: 'phone',
  email: 'email',
  fullName: 'fullName',
  platformUserId: 'platformUserId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.CustomerOrganizationScalarFieldEnum = {
  id: 'id',
  customerId: 'customerId',
  organizationId: 'organizationId',
  loyaltyPoints: 'loyaltyPoints',
  notes: 'notes',
  tags: 'tags',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SchedulableResourceScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  type: 'type',
  name: 'name',
  capacity: 'capacity',
  isActive: 'isActive',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.CentralBookingScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  customerId: 'customerId',
  type: 'type',
  startTime: 'startTime',
  endTime: 'endTime',
  status: 'status',
  timezone: 'timezone',
  metadata: 'metadata',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.BookingResourceLinkScalarFieldEnum = {
  id: 'id',
  bookingId: 'bookingId',
  resourceId: 'resourceId'
};

exports.Prisma.HotelBookingDetailsScalarFieldEnum = {
  bookingId: 'bookingId',
  roomId: 'roomId',
  guestCount: 'guestCount',
  nightCount: 'nightCount'
};

exports.Prisma.SalonBookingDetailsScalarFieldEnum = {
  bookingId: 'bookingId',
  staffId: 'staffId',
  serviceIds: 'serviceIds',
  durationMinutes: 'durationMinutes'
};

exports.Prisma.RestaurantReservationDetailsScalarFieldEnum = {
  bookingId: 'bookingId',
  tableId: 'tableId',
  guestCount: 'guestCount'
};

exports.Prisma.CouponScalarFieldEnum = {
  id: 'id',
  organizationId: 'organizationId',
  code: 'code',
  type: 'type',
  value: 'value',
  minOrderValue: 'minOrderValue',
  maxDiscount: 'maxDiscount',
  startDate: 'startDate',
  endDate: 'endDate',
  usageLimit: 'usageLimit',
  perUserLimit: 'perUserLimit',
  isActive: 'isActive',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.CouponScopeScalarFieldEnum = {
  id: 'id',
  couponId: 'couponId',
  moduleType: 'moduleType',
  entityType: 'entityType'
};

exports.Prisma.CouponRedemptionScalarFieldEnum = {
  id: 'id',
  couponId: 'couponId',
  userId: 'userId',
  organizationId: 'organizationId',
  referenceId: 'referenceId',
  discountApplied: 'discountApplied',
  createdAt: 'createdAt'
};

exports.Prisma.NotificationScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  title: 'title',
  message: 'message',
  isRead: 'isRead',
  createdAt: 'createdAt'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.NullableJsonNullValueInput = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};

exports.Prisma.JsonNullValueFilter = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull,
  AnyNull: Prisma.AnyNull
};


exports.Prisma.ModelName = {
  Organization: 'Organization',
  Brand: 'Brand',
  User: 'User',
  Hotel: 'Hotel',
  Room: 'Room',
  Booking: 'Booking',
  Website: 'Website',
  CMSPage: 'CMSPage',
  CMSSection: 'CMSSection',
  Restaurant: 'Restaurant',
  Table: 'Table',
  Menu: 'Menu',
  MenuCategory: 'MenuCategory',
  MenuItem: 'MenuItem',
  Order: 'Order',
  OrderItem: 'OrderItem',
  InventoryItem: 'InventoryItem',
  InventoryTransaction: 'InventoryTransaction',
  Staff: 'Staff',
  Attendance: 'Attendance',
  GuestProfile: 'GuestProfile',
  Invoice: 'Invoice',
  Expense: 'Expense',
  ActivityLog: 'ActivityLog',
  ParlorCategory: 'ParlorCategory',
  ParlorService: 'ParlorService',
  ServiceBooking: 'ServiceBooking',
  BookingServiceItem: 'BookingServiceItem',
  PlatformUser: 'PlatformUser',
  OrganizationMembership: 'OrganizationMembership',
  Customer: 'Customer',
  CustomerOrganization: 'CustomerOrganization',
  SchedulableResource: 'SchedulableResource',
  CentralBooking: 'CentralBooking',
  BookingResourceLink: 'BookingResourceLink',
  HotelBookingDetails: 'HotelBookingDetails',
  SalonBookingDetails: 'SalonBookingDetails',
  RestaurantReservationDetails: 'RestaurantReservationDetails',
  Coupon: 'Coupon',
  CouponScope: 'CouponScope',
  CouponRedemption: 'CouponRedemption',
  Notification: 'Notification'
};

/**
 * This is a stub Prisma Client that will error at runtime if called.
 */
class PrismaClient {
  constructor() {
    return new Proxy(this, {
      get(target, prop) {
        let message
        const runtime = getRuntime()
        if (runtime.isEdge) {
          message = `PrismaClient is not configured to run in ${runtime.prettyName}. In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters
`;
        } else {
          message = 'PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `' + runtime.prettyName + '`).'
        }

        message += `
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report`

        throw new Error(message)
      }
    })
  }
}

exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
